backend/db/queries/orderQueries.js

```diff
@@ -32,7 +32,71 @@ const deleteOrder = (req, res, next) => {
     });
 };
 
+// const getAllOrders = (req, res, next) => {
+//   db.any(
+//     `SELECT orders.id AS order_id, users.id AS user_id, dishes.id AS dish_id, dishes.user_id AS grandma_id FROM orders JOIN users ON users.id = orders.user_id JOIN dishes ON dishes.id = orders.dish_id`
+//   )
+//     .then(orders => {
+//       res.status(200).json({
+//         status: "success",
+//         orders,
+//         message: "all orders"
+//       });
+//     })
+//     .catch(err => next(err));
+// };
+
+const getAllOrderForSingleUser = (req, res, next) => {
+  const user_id = Number(req.params.user_id);
+  db.any(
+    `SELECT orders.id AS order_id, users.id AS user_id, dishes.id AS dish_id, dishes.user_id AS grandma_id FROM orders JOIN users ON users.id = orders.user_id JOIN dishes ON dishes.id = orders.dish_id
+    WHERE users.id = $1`,
+    [user_id]
+  )
+    .then(orders => {
+      res.status(200).json({
+        status: "success",
+        orders,
+        message: "all orders for a single user"
+      });
+    })
+    .catch(err => next(err));
+};
+
+const allOrdersForGrandma = (req, res, next) => {
+  const grandma_id = Number(req.params.grandma_id);
+  db.any(
+    `SELECT orders.id AS order_id, users.id AS user_id, dishes.id AS dish_id, dishes.user_id AS grandma_id FROM orders JOIN users ON users.id = orders.user_id JOIN dishes ON dishes.id = orders.dish_id WHERE dishes.user_id = $1`,
+    [grandma_id]
+  )
+    .then(order => {
+      res.status(200).json({
+        status: "success",
+        order,
+        message: "single order for user"
+      });
+    })
+    .catch(err => next(err));
+};
+
+const orderSummary = (req, res, next) => {
```

Comment:
I've realized now a way to explain better why assigning res.status in the database layer (DAO) is a bad idea.

1. imagine you have two common queries and you want to perform them side by side instead of merging them into a single query through some `JOIN` black magic.
2.Imagine also you have a route which needs to perform a query AND perform an unrelated network request (whose response you need for your response). If the query sets the `res` response object you are toast. If instead the query just returns the results of the query you can perform the network request asynchronously and once both have happened you would use what parts you need of the both of them in the response.

Basically this makes it impossible for you to reuse the query functions, they MUST be the only thing a route calls, making them tightly coupled to the routes. This is why a separation of concerns is good, because it gives you the options and capability for reuse.